src/Syncs/IBLTSync_Adaptive.cpp

```diff
@@ -0,0 +1,254 @@
+            currentExpected = currentExpected * 2 - totalDecoded;
+            cout << "[Client]: Current Size: " << currentExpected << std::endl;
+        } else {
+            mySyncStats.timerStart(SyncStats::COMM_TIME);
+            list<shared_ptr<DataObject>> newOMS = commSync->commRecv_DataObject_List();
+            list<shared_ptr<DataObject>> newSMO = commSync->commRecv_DataObject_List();
+            mySyncStats.timerEnd(SyncStats::COMM_TIME);
+
+            mySyncStats.timerStart(SyncStats::COMP_TIME);
+            otherMinusSelf.insert(otherMinusSelf.end(), newOMS.begin(), newOMS.end());
+            selfMinusOther.insert(selfMinusOther.end(), newSMO.begin(), newSMO.end());
+            mySyncStats.timerEnd(SyncStats::COMP_TIME);
+
+            mySyncStats.increment(SyncStats::XMIT, commSync->getXmitBytes());
+            mySyncStats.increment(SyncStats::RECV, commSync->getRecvBytes());
+            return true;
+        }
+    }
+}
+        // insert the decoded elements and record Stats
+//        if (success) {
+//            mySyncStats.timerStart(SyncStats::COMM_TIME);
+//            list<shared_ptr<DataObject>> newOMS = commSync->commRecv_DataObject_List();
+//            list<shared_ptr<DataObject>> newSMO = commSync->commRecv_DataObject_List();
+//            mySyncStats.timerEnd(SyncStats::COMM_TIME);
+//
+//            mySyncStats.timerStart(SyncStats::COMP_TIME);
+//            otherMinusSelf.insert(otherMinusSelf.end(), newOMS.begin(), newOMS.end());
+//            selfMinusOther.insert(selfMinusOther.end(), newSMO.begin(), newSMO.end());
+//            mySyncStats.timerEnd(SyncStats::COMP_TIME);
+//
+//            mySyncStats.increment(SyncStats::XMIT, commSync->getXmitBytes());
+//            mySyncStats.increment(SyncStats::RECV, commSync->getRecvBytes());
+//
+//            return true;
+//        } else {
+//            Logger::gLog(Logger::METHOD_DETAILS, "Sync failed. Doubling IBLT size to " + toStr(currentExpected * 2));
+//            currentExpected *= 2;
+// //            commSync->commClose();
+//        }
+
+bool IBLTSync_Adaptive::SyncServer(const shared_ptr<Communicant>& commSync,
+                                   list<shared_ptr<DataObject>> &selfMinusOther,
+                                   list<shared_ptr<DataObject>> &otherMinusSelf) {
+    Logger::gLog(Logger::METHOD, "Entering IBLTSync_Adaptive::SyncServer");
+    size_t currentExpected = initExpNumElems;
+
+
+    // call parent method for bookkeeping
+    SyncMethod::SyncServer(commSync, selfMinusOther, otherMinusSelf);
+
+    // listen for client
+    mySyncStats.timerStart(SyncStats::IDLE_TIME);
+    commSync->commListen();
+    mySyncStats.timerEnd(SyncStats::IDLE_TIME);
+
+    while (true) {
+        currentExpected = static_cast<size_t>(commSync->commRecv_int());
+
+        // construct new IBLT with updated size
+        myIBLT = IBLT::Builder()
```

Comment:
see comment before ... you want these in one common procedure so that if you make changes one place, they get reflected elsewhere.